src/tools/cgset.c

```diff
@@ -163,17 +219,85 @@ static int cgroup_copy_controller_idx(struct cgroup *dst_cgrp, struct cgroup *sr
 	return 0;
 }
 
+static int cgroup_populate_cgroup_ctrl(struct cgroup *dst, const char * const new_cgrp)
```

Comment:
The caller passes a NULL pointer into `dst`, and `dst` is not returned to the caller.  (I fell into this rabbit hole because I was afraid of a memory leak from the caller of this function.)

Wouldn't it be simpler/better if `dst` were a local variable:
```
static int cgroup_populate_cgroup_ctrl(const char * const new_cgrp)
{
...
        struct cgroup *dst;
....
```

Reply:
@drakenclimber Thank you so much for the review.  Agree, will remove it from the argument list.